### PR TITLE
Daml-LF: Add  CAST_NUMERIC and SHIFT_NUMERIC

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR advances the Numeric implementation #2289.

This PR introduces the Builtins CAST_NUMERIC and SHIFT_NUMERIC in the daml-lf side as describeb in #2289.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [X] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/spec/daml-lf-1.rst [hunk 1]
```diff
@@ -2148,6 +2148,19 @@ Numeric functions
...
   `α` inclusive.
 
+
+* ``CAST_NUMERIC : ∀ (α₁, α₂: nat) . 'Numeric' α₁ → 'Numeric' α₂``
+
+  Converts a decimal of scale `α₁` to a decimal scale `α₂` while
+  keeping the value the same. Throws an exception in case of
+  overflow or precision loss.
+
+* ``SHIFT_NUMERIC : ∀ (α₁, α₂: nat) . 'Int64' → 'Numeric' α → 'Numeric' α``
```
Comment (associahedron) [inline daml-lf/spec/daml-lf-1.rst]:
```suggestion
* ``SHIFT_NUMERIC : ∀ (α₁, α₂: nat) . 'Int64' → 'Numeric' α₁ → 'Numeric' α₂``
```

File: daml-lf/archive/da/daml_lf_1.proto [hunk 2]
```diff
@@ -343,6 +343,8 @@ enum BuiltinFunction {
   MUL_NUMERIC = 109;    // *Available in versions >= 1.dev*
   DIV_NUMERIC = 110;    // *Available in versions >= 1.dev*
   ROUND_NUMERIC = 111;  // *Available in versions >= 1.dev*
+  CAST_NUMERIC = 121;   // *Available in versions >= 1.dev*
+  SHIFT_NUMERIC = 123;  // *Available in versions >= 1.dev*
```
Comment (remyhaemmerle-da) [inline daml-lf/archive/da/daml_lf_1.proto]:
```suggestion
  SHIFT_NUMERIC = 122;  // *Available in versions >= 1.dev*
```

[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/archive/da/daml_lf_1.proto]: Indeed. Thanks.

**Comment by hurryabit** [comment]
Nice. Thanks.

**Pull request merged**